pkg/controller/node/status.go

```diff
@@ -0,0 +1,179 @@
+package node
+
+import (
+	"fmt"
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (ctrl *Controller) syncStatusOnly(pool *mcfgv1.MachineConfigPool) error {
+	selector, err := metav1.LabelSelectorAsSelector(pool.Spec.MachineSelector)
+	if err != nil {
+		return err
+	}
+	nodes, err := ctrl.nodeLister.List(selector)
+	if err != nil {
+		return err
+	}
+
+	newStatus := calculateStatus(pool, nodes)
+	if equality.Semantic.DeepEqual(pool.Status, newStatus) {
+		return nil
+	}
+
+	newPool := pool
+	newPool.Status = newStatus
+	_, err = ctrl.client.MachineconfigurationV1().MachineConfigPools(newPool.Namespace).UpdateStatus(newPool)
+	return err
+}
+
+func calculateStatus(pool *mcfgv1.MachineConfigPool, nodes []*corev1.Node) mcfgv1.MachineConfigPoolStatus {
+	machineCount := int32(len(nodes))
+
+	updatedMachines := getUpdatedMachines(pool.Status.CurrentMachineConfig, nodes)
+	updatedMachineCount := int32(len(updatedMachines))
+
+	readyMachines := getReadyMachines(pool.Status.CurrentMachineConfig, nodes)
+	readyMachineCount := int32(len(readyMachines))
+
+	unavailableMachines := getUnavailableMachines(pool.Status.CurrentMachineConfig, nodes)
+	unavailableMachineCount := int32(len(unavailableMachines))
+
+	degradedMachines := getDegradedMachines(pool.Status.CurrentMachineConfig, nodes)
+
+	status := mcfgv1.MachineConfigPoolStatus{
+		ObservedGeneration:      pool.Generation,
+		MachineCount:            machineCount,
+		UpdatedMachineCount:     updatedMachineCount,
+		ReadyMachineCount:       readyMachineCount,
+		UnavailableMachineCount: unavailableMachineCount,
+	}
+
+	status.CurrentMachineConfig = pool.Status.CurrentMachineConfig
+
+	conditions := pool.Status.Conditions
+	for i := range conditions {
+		status.Conditions = append(status.Conditions, conditions[i])
+	}
+
+	if updatedMachineCount == machineCount &&
+		readyMachineCount == machineCount &&
+		unavailableMachineCount == 0 {
+		supdated := mcfgv1.NewMachineConfigPoolCondition(mcfgv1.MachineConfigPoolUpdated, corev1.ConditionTrue, fmt.Sprintf("All nodes are updated with %s", pool.Status.CurrentMachineConfig), "")
```

Comment:
why it's setting two states at the same time?

Reply:
added a todo, didn't wan to update the api in this PR.

Reply:
sgtm.